react/rules/BRA.js

```diff
@@ -34,40 +34,51 @@ export default {
       label: 'BRA_cpf',
       required: true,
       mask: value => msk.fit(value, '999.999.999-99'),
-      validate: value => {
-        const cleanValue = value.replace(/[^\d]/g, '')
-
-        if (cleanValue.length !== 11) return false
-
-        const isRepeatedNum = '0123456789'
-          .split('')
-          .some(digit => digit.repeat(11) === cleanValue)
-
-        if (isRepeatedNum) return false
-
-        const firstReduce = cleanValue
-          .split('')
-          .slice(0, 9)
-          .reduce(
-            (acc, cur, index) => acc + parseInt(cur, 10) * (10 - index),
-            0
-          )
-
-        const firstDigit = ((firstReduce * 10) % 11) % 10
-
-        if (firstDigit !== cleanValue.charAt(9)) return false
-
-        const secondReduce = cleanValue
-          .split('')
-          .slice(0, 10)
-          .reduce(
-            (acc, cur, index) => acc + parseInt(cur, 10) * (11 - index),
-            0
-          )
-
-        const secondDigit = ((secondReduce * 10) % 11) % 10
-
-        return secondDigit === cleanValue.charAt(10)
+      validate: cpf => {
+        if (!cpf) {
+          return false
+        }
+
+        let sum = 0
+        let remainder
+
+        cpf = !cpf || typeof cpf !== 'string' ? '' : cpf.replace(/[^\d]+/g, '')
+
+        if (
+          cpf === '00000000000' ||
+          cpf === '11111111111' ||
+          cpf === '22222222222' ||
+          cpf === '33333333333' ||
+          cpf === '44444444444' ||
+          cpf === '55555555555' ||
+          cpf === '66666666666' ||
+          cpf === '77777777777' ||
+          cpf === '88888888888' ||
+          cpf === '99999999999'
+        ) {
+          return false
+        }
+
+        for (let i = 1; i <= 9; i++) {
+          sum += parseInt(cpf.substring(i - 1, i), 10) * (11 - i)
+        }
+
+        remainder = (sum * 10) % 11
+
+        if (remainder === 10 || remainder === 11) remainder = 0
+        if (remainder !== parseInt(cpf.substring(9, 10), 10)) return false
+
+        sum = 0
```

Comment:
every time that you had the needed to re-assign a variable its a indication that you should think in encapsulate the code above this re-assign

Reply:
On my new commit I didn't re-assign that variable, so it shouldn't be a problem. But I didn't understand what you mean by 'encapsulate the code'